lib/shark.rb

```diff
@@ -34,26 +34,45 @@ def self.configure
   # @api public
   def self.with_service_token(token)
     if token.is_a?(String)
-      self.service_token = token
+      self.auth_token = "Bearer #{token}"
     elsif token.respond_to?(:jwt)
-      self.service_token = token.jwt
+      self.auth_token = "Bearer #{token.jwt}"
     else
       raise ArgumentError, 'Parameter :token must be kind of String.'
     end
 
     yield
   ensure
-    self.service_token = nil
+    self.auth_token = nil
+  end
```

Comment:
I suggest to remove code duplication and make use of your more general `#with_auth_token` function here.

```
  def self.with_service_token(token)
    if token.is_a?(String)
      auth_token = "Bearer #{token}"
    elsif token.respond_to?(:jwt)
      auth_token = "Bearer #{token.jwt}"
    end

    with_auth_token(auth_token)
  end
```